src/Simplex/Messaging/Protocol.hs

```diff
@@ -336,6 +340,8 @@ data BrokerMsg where
   MSG :: RcvMessage -> BrokerMsg
   NID :: NotifierId -> RcvNtfPublicDhKey -> BrokerMsg
   NMSG :: C.CbNonce -> EncNMsgMeta -> BrokerMsg
+  RKEY :: C.PublicKeyX25519 -> C.Signature C.Ed25519 -> BrokerMsg -- TLS-signed server key for proxy shared secret and initial sender key
```

Comment:
```suggestion
  RKEY :: C.PublicKeyX25519 -> Cert (?) -> C.Signature C.Ed25519 -> BrokerMsg -- TLS-signed server key for proxy shared secret and initial sender key. Sent by relay to proxy
  PRKEY :: C.PublicKeyX25519 -> Cert -> C.Signature C.Ed25519 -> BrokerMsg - sent by proxy to client.
```

I am not sure how this signature can be validated here, as we do not have the key, only fingerprint of the certificate, so possibly the entire certificate that the relay received in TLS handshake (two-cert self-signed chain) should be included in this command. Then we can validate the signature.